packages/netlify-cms-backend-github/src/API.js

```diff
@@ -500,47 +505,65 @@ export default class API {
     });
   }
 
-  deleteFile(path, message, options = {}) {
-    const branch = options.branch || this.branch;
+  getFileSha(path, branch) {
+    /**
+     * We need to request the tree first to get the SHA. We use extended SHA-1
+     * syntax (<rev>:<path>) to get a blob from a tree without having to recurse
+     * through the tree.
+     */
+
     const pathArray = path.split('/');
     const filename = last(pathArray);
     const directory = initial(pathArray).join('/');
     const fileDataPath = encodeURIComponent(directory);
     const fileDataURL = `${this.repoURL}/git/trees/${branch}:${fileDataPath}`;
-    const fileURL = `${this.repoURL}/contents/${path}`;
 
-    /**
-     * We need to request the tree first to get the SHA. We use extended SHA-1
-     * syntax (<rev>:<path>) to get a blob from a tree without having to recurse
-     * through the tree.
-     */
     return this.request(fileDataURL, { cache: 'no-store' }).then(resp => {
       const { sha } = resp.tree.find(file => file.path === filename);
+      return sha;
+    });
+  }
+
+  deleteFile(path, message, options = {}) {
+    const branch = options.branch || this.branch;
+
+    return this.getFileSha(path, branch).then(sha => {
       const opts = { method: 'DELETE', params: { sha, message, branch } };
       if (this.commitAuthor) {
         opts.params.author = {
           ...this.commitAuthor,
           date: new Date().toISOString(),
         };
       }
+      const fileURL = `${this.repoURL}/contents/${path}`;
       return this.request(fileURL, opts);
     });
   }
 
+  async createBranchAndPullRequest(branchName, sha, commitMessage) {
+    await this.createBranch(branchName, sha);
+    if (this.useForkWorkflow) {
+      return undefined;
+    }
+    return this.createPR(commitMessage, branchName);
+  }
+
   async editorialWorkflowGit(fileTree, entry, filesList, options) {
     const contentKey = this.generateContentKey(options.collectionName, entry.slug);
     const branchName = this.generateBranchName(contentKey);
     const unpublished = options.unpublished || false;
-    const branchData = await this.getBranch();
```

Comment:
BUG: Looks like a bug introduced in https://github.com/netlify/netlify-cms/pull/2430

This breaks updating an existing entry since instead of rebasing the pull request branch it will try to rebase master

Reply:
Fixed in #2523: https://github.com/netlify/netlify-cms/pull/2523/commits/aa664adf8794a3050af592c952388b71b9ee25e4